### DISCUSSION TITLE
Automatic Cleaning + Firefox Container Tabs

### DISCUSSION THREAD

**Discussion by Bout2GitIt** [Cookie-AutoDelete/Cookie-AutoDelete#1128] [Q&A / Support] [answered]
The CAD Settings menu suggests that using the Automatic Cleaning function with Container Tabs is undesirable.
("WARNING: Enabling Container Tabs with any of the additional browsing data cleanups above may cause unwanted side effects" Automatic Cleaning is listed above this message.)

Am I reading this correctly? Do we have to choose between using CAD's Automatic Delete function and Firefox's Container Tabs?

I ask because I've been experiencing issues lately (crashed tabs, freezing) that might be related to CAD and Container Tabs (maybe even related to my use of an add-on called Google Container, which is a fork of Mozilla's official Facebook Container add-on). I've turned off Automatic Cleaning in CAD and that seems to have addressed the worst of it, but... is CAD doing anything now? If it's not automatically deleting cookies, do I need to keep it?

**Comment by kennethtran93** [comment, marked as answer]
I don't think you read it correctly. The warning applies to any of the 5 cleanup types mentioned in the screenshot below when using containers.
![image](https://user-images.githubusercontent.com/6724477/116145183-d7726800-a691-11eb-8b3c-992b7971189c.png).

It works fine when only cookies are removed using the non-strict privacy setting.
If you are experiencing freezing, disable the cache cleanup option.  Seems Firefox's cleanup implementation for that isn't as quick as the rest.

To slightly elaborate more on the warning shown:
Issues may arise when you attempt to clean any of the other browsing data cleanup types with containers.  As of now, if at least one domain requires cleanup of one of those other types in one container but not in another container, it will be cleaned up in ALL containers.  This is not a CAD bug, but the Firefox WebExtension API one.

**Reply by kennethtran93** [reply]
Adding another note that if it is freezing, it may have to do with the cache cleanup.  This is also a Firefox bug as well.
